Issue 34519 allow admin to modify user profile picture

Kishn0109 opened the pull request:
<!-- Describe your pull request here.-->

Fixes: https://github.com/zulip/zulip/issues/34519

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

## Manage User Profile in Desktop:
| Before | After |
| --- | --- |
| <img width="596" height="673" alt="image" src="https://github.com/user-attachments/assets/ea5517b9-e4e0-4e9a-b7b1-e290b49f01be" /> | <img width="664" alt="image" src="https://github.com/user-attachments/assets/1f3c237b-d0dd-4108-ba2e-518d1e19398b" /> | 


## Manage User Profile in Small Screen:
| Before | After |
| --- | --- |
| <img width="474" height="668" alt="image" src="https://github.com/user-attachments/assets/23690db3-27cb-4e63-9427-bc47e575f63e" /> | <img width="553" alt="image" src="https://github.com/user-attachments/assets/d462a872-9eed-4ceb-bba4-5dcb940c5596" /> | 






<img width="592" height="178" alt="image" src="https://github.com/user-attachments/assets/401c3346-d26c-4dd2-b5ca-6a404bd93b82" />





<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

Kishn0109 commented:
https://github.com/user-attachments/assets/9ddb676e-7e19-40f6-aa04-1c17db57a2d3


- [x] Added the profile picture to the "manage user" panel in profile settings.
- [x] Added the same type of avatar upload UI as in personal settings.

alya commented:
Thanks! Please clean up your commit history and post again to request a review. See [here](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for guidelines.

You will also need to include still screenshots of all your changes, and make sure the PR passes tests.

alya commented:
Thanks, but the commit history still does not follow the guidelines, and the PR is not passing test, which are prerequisites for code review.

Kishn0109 commented:
All tests are passing now, and I’ve checked all commits with ./tools/commit-message-lint—no errors found. Could you please take a look for the first level of code review?

alya commented:
The commit history does not follow the guidelines linked above.

Kishn0109 commented:
Hi @alya, thanks for getting back to me. I’ve updated the commit history — let me know if anything else needs changing.

Kishn0109 commented:
Hi @alya, I’ve cleaned up the commit history into a single well-structured commit with a clear message documenting the changes. Could you please take another look and let me know if it’s good for review?

alya commented:
Is the "Delete profile picture?" confirmation in your screenshot shown to admins? If so, it will need to say "this profile picture", rather than "your profile picture", when it's a different user.

alya commented:
It would be helpful to clearly label your screenshots, and more comprehensively capture the states. Think of going trough the video you posted and showing key points as still images, in order.

Kishn0109 committed:
avatar: Add admin support for avatar management.

This commit introduces a complete avatar management system.

- Adds API endpoints for uploading, managing, and deleting avatars.
- Extends backend support to track avatar_source for each user.
- Implements admin controls and permission checks for avatar changes.
- Updates type safety for avatar handling code paths.
- Extends tests and fixtures to include the new avatar_source field.
- Documents the new avatar_source field in the OpenAPI schema for
  relevant user endpoints.

Previously, avatar handling was limited and lacked clear source
tracking. With this change, both users and admins can manage avatars
more reliably, and developers have a consistent type-safe API for
avatar operations.

Kishn0109 commented:
Thanks for the feedback — I’ve updated the confirmation text and added clearer, labeled screenshots covering the key states.

alya commented:
The vertical alignment of the profile image in this tab should match the profile tab. Ideally horizontal alignment too, since then you wouldn't see the image move around when you switch tabs.

alya commented:
Can we keep the profile modal open when doing the picture deletion confirmation?

Kishn0109 commented:
@alya I have fixed the image resolution and made the avatar width consistent across both tabs.

| Profile Tab | Manage User Tab |
| --- | --- |
| <img width="644" height="629" alt="image" src="https://github.com/user-attachments/assets/bd148650-6632-4c84-97d2-5f450fd50eb2" /> | <img width="644" height="629" alt="image" src="https://github.com/user-attachments/assets/f2d66fe8-98f5-414e-b8c9-7d8f2b3ac264" /> |
